### PR TITLE
Rspec3 syntax

### PR DESCRIPTION
**Pull request by listrophy**
Anyone who's writing their own adapter and using flipper's `shared_examples` to test it are likely to get deprecation warnings.

This PR includes 2 unrelated things... sorry!

**First** (and most importantly), it switches the `should` syntax to `expect`. Nearly all the conversion was done with http://xinminlabs.github.io/synvert/, so there should be less chance for user error. The only manually converted file is the shared_spec file in lib.

**Second**, I got a bit frustrated running specs on a clean repo the first time. Turns out, the mongo adapter waits 60 (or so) seconds (per spec!) before failing if it can't connect to the mongo server. This PR shortens that to 1 second and outputs a helpful message about maybe mongo not running.

### CODE REVIEW

File: spec/flipper/adapters/mongo_spec.rb [hunk 1]
```diff
@@ -16,8 +17,19 @@
 
   before do
     begin
-      collection.drop
-    rescue Mongo::Error::OperationFailure
+      Timeout::timeout(1) do
```
Comment (jnunemaker) [inline spec/flipper/adapters/mongo_spec.rb]: Can we just set the timeout in the mongo client initialization and call it good? A rescue would be good, but I'd rather just use their timeout than a timeout like this.

**Comment by jnunemaker** [comment]
Looks good other than one comment I left.

File: spec/flipper/adapters/mongo_spec.rb (hunk 1 above)
Comment (listrophy) [inline spec/flipper/adapters/mongo_spec.rb]: Since it truly is a separate concern, I just reverted the commit related to this. Setting the timeout in the mongo client initialization can be a separate PR.
